### DISCUSSION TITLE
Problem with limited connection

### DISCUSSION THREAD

**Discussion by AgentCousto** [aws/aws-sdk-cpp#2766] [Q&A] [answered]
Hello,
I use the AWS SDK to push files daily to S3.
I have an upload connection close to 100KB/s, and I'm trying to send files of size 50MB.
I don't understand why my files are not uploading I have the following messages in the AWS logs:

[WARN] 2023-11-24 00:11:28.191 WinHttpSyncHttpClient [11144] Send request failed: La connexion avec le serveur a été interrompue anormalement

[ERROR] 2023-11-24 00:11:28.191 AWSClient [11144] HTTP response code: -1
Resolved remote host IP address: 
Request ID: 
Exception name: 
Error message: Encountered network error when sending http request
0 response headers:
[WARN] 2023-11-24 00:11:28.191 AWSClient [11144] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.
[WARN] 2023-11-24 00:11:28.191 AWSClient [11144] Request failed, now waiting 1600 ms before attempting again.
[INFO] 2023-11-24 00:11:30.007 WinHttpConnectionPoolMgr [11144] Attempting to acquire connection for ayro-canopee.s3.eu-west-3.amazonaws.com:443
[INFO] 2023-11-24 00:11:30.007 WinHttpConnectionPoolMgr [11144] Connection now available, continuing.
[WARN] 2023-11-24 00:11:30.007 WinHttpSyncHttpClient [11144] Failed setting TCP keep-alive interval with error code: 12018
[WARN] 2023-11-24 00:12:03.091 WinHttpSyncHttpClient [11144] Send request failed: La connexion avec le serveur a été interrompue anormalement

[ERROR] 2023-11-24 00:12:03.091 AWSClient [11144] HTTP response code: -1
Resolved remote host IP address: 
Request ID: 
Exception name: 
Error message: Encountered network error when sending http request
0 response headers:
[WARN] 2023-11-24 00:12:03.091 AWSClient [11144] If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.
[WARN] 2023-11-24 00:12:03.091 AWSClient [11144] Request failed, now waiting 3200 ms before attempting again.
[INFO] 2023-11-24 00:12:06.508 WinHttpConnectionPoolMgr [11144] Attempting to acquire connection for ayro-canopee.s3.eu-west-3.amazonaws.com:443
[INFO] 2023-11-24 00:12:06.508 WinHttpConnectionPoolMgr [11144] Connection now available, continuing.
[WARN] 2023-11-24 00:12:06.508 WinHttpSyncHttpClient [11144] Failed setting TCP keep-alive interval with error code: 12018

Thanks for your help

**Comment by jmklix** [comment]
Is the time on the machine you are running this correct? Your getting this error:
```
If the signature check failed. This could be because of a time skew. Attempting to adjust the signer.
```
Which suggests that the time might be incorrect. If the time is more than 15 minutes off from what the AWS servers expect on the signatures then this can cause issues. This sdk will try to resolve the errors but it doesn't always work. Can you confirm that you have the correct time (or close to) on your machine. 

If that doesn't work can you provide the following:
 - a minimal reproduction sample
 - C++ sdk version
 - OS version
 - how you are compiling/building this sdk

**Reply by AgentCousto** [reply]
Thank you for your response, the SDK runs on a boat that sails from Europe to America. My PC is running Windows 10 and the clock is in UTC synchronized via the internet. Could it be a time zone problem?

**Reply by jmklix** [reply, marked as answer]
It could possibly be a time zone problem and that is a cool and unique scenario that you are in. Did moving to a different time zone cause this to start failing?
